src/brokercollect/broker_collection.go

```diff
@@ -86,66 +86,75 @@ func brokerWorker(brokerChan <-chan int, collectedTopics []string, wg *sync.Wait
 		}
 
 		// Create Broker
-		b, err := createBroker(brokerID, zkConn, i)
+		brokers, err := createBrokers(brokerID, zkConn, i)
 		if err != nil {
 			continue
 		}
 
-		// Populate inventory for broker
-		if args.GlobalArgs.All() || args.GlobalArgs.Inventory {
-			log.Debug("Collecting inventory for broker %q", b.Entity.Metadata.Name)
-			if err := populateBrokerInventory(b); err != nil {
-				continue
+		for _, broker := range brokers {
```

Comment:
It appears this loop will collect metrics/inventory multiple times if multiple connections succeed